src/Groups/matrices/MatGrp.jl

```diff
@@ -0,0 +1,892 @@
+   AbstractAlgebra.@show_special(io,x)
+
+   if isdefined(x, :descr)
+      if x.descr==:GU || x.descr==:SU
+         print(io, string(x.descr), "(",x.deg,",",characteristic(x.ring)^(div(degree(x.ring),2)),")")
+      else
+         print(io, string(x.descr), "(",x.deg,",",order(x.ring),")")
+      end
+   else
+      print(io, "Matrix group of degree ", x.deg, " over ")
+      show(IOContext(io, :compact => true), x.ring)
+   end
+end
+
+function Base.show(io::IO, x::MatrixGroupElem)
+   if isdefined(x, :elm)
+      show(io, "text/plain", x.elm)
+#      print(io, x.elm)
+   else
+      print(io, GAP.gap_to_julia(GAP.Globals.StringViewObj(x.X)))
+   end
+end
+
+group_element(G::MatrixGroup, x::GapObj) = MatrixGroupElem(G,x)
+
+function assign_from_description(G::MatrixGroup)
+   if G.descr==:GL G.X=GAP.Globals.GL(G.deg,G.ring_iso.codomain)
+   elseif G.descr==:SL G.X=GAP.Globals.SL(G.deg,G.ring_iso.codomain)
+   elseif G.descr==:Sp G.X=GAP.Globals.Sp(G.deg,G.ring_iso.codomain)
+   elseif G.descr==Symbol("GO+") G.X=GAP.Globals.GO(1,G.deg,G.ring_iso.codomain)
+   elseif G.descr==Symbol("SO+") G.X=GAP.Globals.SO(1,G.deg,G.ring_iso.codomain)
+   elseif G.descr==Symbol("Omega+") G.X=GAP.Globals.Omega(1,G.deg,Int(order(G.ring)))
+   elseif G.descr==Symbol("GO-") G.X=GAP.Globals.GO(-1,G.deg,G.ring_iso.codomain)
+   elseif G.descr==Symbol("SO-") G.X=GAP.Globals.SO(-1,G.deg,G.ring_iso.codomain)
+   elseif G.descr==Symbol("Omega-") G.X=GAP.Globals.Omega(-1,G.deg,Int(order(G.ring)))
+   elseif G.descr==:GO G.X=GAP.Globals.GO(0,G.deg,G.ring_iso.codomain)
+   elseif G.descr==:SO G.X=GAP.Globals.SO(0,G.deg,G.ring_iso.codomain)
+   elseif G.descr==:Omega G.X=GAP.Globals.Omega(0,G.deg,Int(order(G.ring)))
+   elseif G.descr==:GU G.X=GAP.Globals.GU(G.deg,Int(characteristic(G.ring)^(div(degree(G.ring),2) ) ))
+   elseif G.descr==:SU G.X=GAP.Globals.SU(G.deg,Int(characteristic(G.ring)^(div(degree(G.ring),2) ) ))
+   else error("unsupported description")
+   end
+end
+
+# return the G.sym if isdefined(G, :sym); otherwise, the field :sym is computed and set using information from other defined fields
+# NOTE: if G.X has to be set, then also the fields G.ring_iso and G.mat_iso are set
+function Base.getproperty(G::MatrixGroup, sym::Symbol)
+
+   if isdefined(G,sym) return getfield(G,sym) end
+
+   if sym === :mat_iso || sym === :ring_iso
+      fm = gen_mat_iso(G.deg, G.ring)
+      G.ring_iso=fm.fr
+      G.mat_iso=fm
+
+   elseif sym === :X
+      if isdefined(G,:descr)
+         if !isdefined(G,:ring_iso)
+            fm = gen_mat_iso(G.deg, G.ring)
+            G.ring_iso=fm.fr
+            G.mat_iso=fm
+         end
+         assign_from_description(G)
+      elseif isdefined(G,:gens)
+         if !isdefined(G,:ring_iso)
+            fm = gen_mat_iso(G.deg, G.ring)
+            G.ring_iso=fm.fr
+            G.mat_iso=fm
+         end
+         V = GAP.julia_to_gap([g.X for g in G.gens])
+         G.X=GAP.Globals.Group(V)
+      end
```

Comment:
Hmmm, and what if neither option holds? Perhaps there should be an `else` case here?
```suggestion
      else
          error("cannot determine underlying GAP group")
      end
```